libs/tests/core/test_toolkit.py

```diff
@@ -0,0 +1,344 @@
+    @patch("arcade_core.toolkit.Toolkit.find_arcade_toolkits_from_prefix")
+    def test_find_all_empty(self, mock_find_prefix, mock_find_ep):
+        """Test when no toolkits are found."""
+        mock_find_ep.return_value = []
+        mock_find_prefix.return_value = []
+
+        toolkits = Toolkit.find_all_arcade_toolkits()
+        assert toolkits == []
+
+
+class TestEntryPointCompatibility:
+    """Test compatibility scenarios for entry point discovery."""
+
+    @patch("arcade_core.toolkit.importlib.metadata.entry_points")
+    @patch("arcade_core.toolkit.Toolkit.from_entrypoint")
+    def test_duplicate_toolkit_names_in_entrypoints(self, mock_from_ep, mock_entry_points):
+        """Test handling of duplicate toolkit names in entry points."""
+        # Create mock entry points with same name
+        mock_ep1 = MagicMock()
+        mock_ep1.name = "toolkit_name"
+        mock_ep1.value = "test_v1"
+
+        mock_ep2 = MagicMock()
+        mock_ep2.name = "toolkit_name"
+        mock_ep2.value = "test_v2"
+
+        mock_entry_points.return_value = [mock_ep1, mock_ep2]
+
+        # Mock toolkit creation - both with same toolkit name
+        toolkit1 = Toolkit(
+            name="test",
+            package_name="test_v1",
+            version="1.0.0",
+            description="Test v1",
+        )
+
+        toolkit2 = Toolkit(
+            name="test",
+            package_name="test_v2",
+            version="2.0.0",
+            description="Test v2",
+        )
+
+        mock_from_ep.side_effect = [toolkit1, toolkit2]
+
+        # Should return both even with same display name
+        toolkits = Toolkit.find_arcade_toolkits_from_entrypoints()
+        assert len(toolkits) == 2
+
+    @patch("arcade_core.toolkit.Toolkit.from_package")
+    def test_from_entrypoint_with_arcade_prefix(self, mock_from_package):
+        """Test that arcade_ prefix is stripped from entry point toolkits."""
+        # Create mock entry point
+        mock_entry = MagicMock()
+        mock_entry.value = "arcade_example"
+        mock_entry.name = "toolkit_name"
+        mock_entry.dist = MagicMock()
+        mock_entry.dist.name = "arcade-example"
+
+        # Mock the from_package return
+        mock_toolkit = Toolkit(
+            name="arcade_example",
+            package_name="arcade-example",
+            version="1.0.0",
+            description="Example toolkit",
+        )
+        mock_from_package.return_value = mock_toolkit
+
+        toolkit = Toolkit.from_entrypoint(mock_entry)
+
+        assert toolkit.name == "example"
+        assert toolkit.package_name == "arcade-example"
+
+
+class TestToolkitIntegration:
+    """Integration tests for toolkit discovery and loading."""
+
+    @patch("arcade_core.toolkit.Toolkit.find_arcade_toolkits_from_entrypoints")
+    @patch("arcade_core.toolkit.Toolkit.find_arcade_toolkits_from_prefix")
+    def test_mixed_toolkit_sources(self, mock_find_prefix, mock_find_ep):
```

Comment:
Nit: could we add a mocked package with the correct prefix but that is not a toolkit?